modules/h12mediaBidAdapter.js

```diff
@@ -16,21 +15,30 @@ export const spec = {
   },
 
   buildRequests: function(validBidRequests, bidderRequest) {
-    const requestUrl = validBidRequests[0].params.endpointdom || DEFAULT_URL;
-    const isiframe = !((window.self === window.top) || window.frameElement);
+    const isiframe = inIframe();
     const screenSize = getClientDimensions();
     const docSize = getDocumentDimensions();
 
-    const bidrequests = validBidRequests.map((bidRequest) => {
+    return validBidRequests.map((bidRequest) => {
       const bidderParams = bidRequest.params;
-      const adUnitElement = document.getElementById(bidRequest.adUnitCode);
+      const requestUrl = bidderParams.endpointdom || DEFAULT_URL;
+      let pubsubid = bidderParams.pubsubid || '';
+      if (pubsubid && pubsubid.length > 32) {
+        utils.logError('Bidder param \'pubsubid\' should be less than 32 chars.');
+        pubsubid = '';
+      }
+      const pubcontainerid = bidderParams.pubcontainerid;
+      const adUnitElement = document.getElementById(pubcontainerid || bidRequest.adUnitCode);
       const ishidden = !isVisible(adUnitElement);
-      const coords = {
+      const coords = isiframe ? {
+        x: getFramePos()[0],
+        y: getFramePos()[1],
```

Comment:
why calling `getFramePos()` twice, why don't you save the returned value in some variable `pos` and use it later as `pos[0]` and `pos[1]`

Reply:
Fixed, added cache variable